test/e2e/main_test.go

```diff
@@ -1,11 +1,49 @@
 package e2e
 
 import (
+	"k8s.io/client-go/kubernetes"
 	"testing"
+	"time"
 
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/tracker"
 	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"k8s.io/api/core/v1"
 )
 
+// testContext holds the information related to the entire test suite. This data structure
+// should be instantiated by every test suite, so that we can update the test context to be
+// passed around to get the information which was created within the test suite. For example,
+// if the create test suite creates a Windows Node object, the node object and other related
+// information should be easily accessible by other methods within the same test suite.
+// Some of the fields we have here can be exposed by via flags
+// to the test suite.
+type testContext struct {
+	// numberOfNodes to be used for the test suite
+	numberOfNodes int
+	// osdkTestCtx holds the information related to operator sdk framework's test Context
+	osdkTestCtx *framework.TestCtx
+	// nodes holds the list of nodes
+	nodes []v1.Node
+	// credentials to be used to the node.
+	credentials      []tracker.Credentials
+	retryInterval    time.Duration
+	timeout          time.Duration
+	kubeclient       kubernetes.Interface
+	trackerConfigMap *v1.ConfigMap
+	//TODO: Add deletionCount as well.
```

Comment:
Do you have a Jira story for this? And is this really needed?

Reply:
I think this will be beneficial but not needed immediately.

Reply:
OK, looking at the comment I don't understand what you want done here, which is why I was asking for a Jira story.